### PR TITLE
RTC-5104: screensharing indicator

### PR DESCRIPTION
**Pull request by sergey-su**
## Description
Implement screensharing indicator banner similar to Chrome one.
(https://perzoinc.atlassian.net/browse/RTC-5104)

## Solution Approach
New JS API creates a frameless, transparent, draggable window.
`showScreenSharingIndicator({stream, displayId}, callback)`.
`stream` is [https://developer.mozilla.org/en-US/docs/Web/API/MediaStream](MediaStream). Banner deletes itself automatically when stream deactivates. `displayId` is a hint - which display to attach banner to. display_id is a part of [DesktopCapturerSource](https://electronjs.org/docs/api/structures/desktop-capturer-source#desktopcapturersource-object) returned by `getSource()`. `callback` is for errors and to escalate "Stop Sharing" button.

## Screen
screen shot 2018-12-18 at 09 17 20

## Related PRs
https://github.com/SymphonyOSF/SFE-RTC/pull/1488

### CODE REVIEW

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community! We require contributors to sign a [Contributor License Agreement](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/75530375/Legal+Requirements#LegalRequirements-ContributorLicenseAgreement) and we don't seem to have CLAs on file for these contributors to the Pull Request: (@sergey-su). In order for your PR to be reviewed and merged, please follow the directions at the link above.

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If there are any questions, please don't hesitate to [get in touch with our Infrastructure Support team](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/77955298/Getting+in+Touch#GettinginTouch-InfrastructureSupport). /CC @finos-admin

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community! We require contributors to sign a [Contributor License Agreement](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/75530375/Legal+Requirements#LegalRequirements-ContributorLicenseAgreement) and we don't seem to have CLAs on file for these contributors to the Pull Request: (@sergey-su). In order for your PR to be reviewed and merged, please follow the directions at the link above.

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If there are any questions, please don't hesitate to [get in touch with our Infrastructure Support team](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/77955298/Getting+in+Touch#GettinginTouch-InfrastructureSupport). /CC @finos-admin

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community! We require contributors to sign a [Contributor License Agreement](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/75530375/Legal+Requirements#LegalRequirements-ContributorLicenseAgreement) and we don't seem to have CLAs on file for these contributors to the Pull Request: (@sergey-su). In order for your PR to be reviewed and merged, please follow the directions at the link above.

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If there are any questions, please don't hesitate to [get in touch with our Infrastructure Support team](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/77955298/Getting+in+Touch#GettinginTouch-InfrastructureSupport). /CC @finos-admin

File: js/screenSharingIndicator/index.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+const baseWindowConfig = {
+    width: 580,
+    height: 45,
+    show: false,
+    modal: true,
+    frame: false,
+    focusable: false,
+    transparent: true,
+    autoHideMenuBar: true,
```
[the remark this answers is not in the document]
Comment (sergey-su) [reply js/screenSharingIndicator/index.js]: Fixed

**Comment by sergey-su** [comment]
Pushed an update with translations to JA and FR

**Comment by maoo** [comment]
@finos-cla-bot[bot] check

**Comment by finos-admin** [comment]
The cla-bot has been summoned, and re-checked this pull request!

File: js/screenSharingIndicator/index.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        devTools: false
+    }
+};
+
+function getTemplatePath() {
+    let templatePath = path.join(__dirname, 'screen-sharing-indicator.html');
+    try {
+        fs.statSync(templatePath).isFile();
+    } catch (err) {
+        log.send(logLevels.ERROR, 'screen-sharing-indicator: Could not find template ("' + templatePath + '").');
```
Comment (VishwasShashidhar) [inline js/screenSharingIndicator/index.js]: Please update this code to use ES6 `Template literals`

File: js/screenSharingIndicator/index.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    }
+    return 'file://' + templatePath;
```
Comment (VishwasShashidhar) [inline js/screenSharingIndicator/index.js]: Please update this code to use ES6 Template literals

File: js/screenSharingIndicator/index.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        electron.dialog.showMessageBox(errorDialogOptions, () => indicatorWindow.close());
+    });
+
+    const handleStopSharingClicked = (event, indicatorId) => {
+        if (indicatorId === id) {
+            eventSender.send('stop-sharing-requested', id);
+        }
+    }
+
+    const handleDestroyScreenshingIndicator = (event, indicatorId) => {
```
Comment (VishwasShashidhar) [inline js/screenSharingIndicator/index.js]: typo in the method name or purposeful?

File: js/screenSharingIndicator/index.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        if (indicatorId === id) {
+            if (!indicatorWindow.isDestroyed()) {
+                indicatorWindow.close();
+            }
+            ipcMain.removeListener('stop-sharing-clicked', handleStopSharingClicked);
+            ipcMain.removeListener('destroy-screenshing-indicator', handleDestroyScreenshingIndicator);
```
Comment (VishwasShashidhar) [inline js/screenSharingIndicator/index.js]: typo in the event & method name or purposeful?

File: js/screenSharingIndicator/index.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        }
+    };
+
+    ipcMain.on('stop-sharing-clicked', handleStopSharingClicked);
+    ipcMain.on('destroy-screenshing-indicator', handleDestroyScreenshingIndicator);
```
Comment (VishwasShashidhar) [inline js/screenSharingIndicator/index.js]: typo in the event & method name or purposeful?

File: js/screenSharingIndicator/index.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+module.exports = {
+    openScreenSharingIndicator
```
Comment (VishwasShashidhar) [inline js/screenSharingIndicator/index.js]: typo in the method name or purposeful?

File: js/screenSharingIndicator/showScreenSharingIndicator.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const { ipcRenderer } = require('electron');
+const apiEnums = require('../enums/api.js');
+const apiCmds = apiEnums.cmds;
+const apiName = apiEnums.apiName;
+
+let nextnextIndicatorId = 0;
```
Comment (VishwasShashidhar) [inline js/screenSharingIndicator/showScreenSharingIndicator.js]: typo?

File: js/screenSharingIndicator/index.js (hunk 7 above)
Comment (sergey-su) [reply js/screenSharingIndicator/index.js]: This name is ok

**Comment by sergey-su** [comment]
Thank you @VishwasShashidhar , fixed

**Comment by VishwasShashidhar** [comment]
@sergey-su All done. I'll create a new build with this branch, let's test it out, we can then merge this PR.

**Comment by sergey-su** [comment]
Verified on win
win


Ok to merge.

**Pull request merged**